sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -239,7 +239,7 @@ object JdbcUtils extends Logging {
       case java.sql.Types.TIMESTAMP     => TimestampType
       case java.sql.Types.TIMESTAMP_WITH_TIMEZONE
                                         => null
-      case java.sql.Types.TINYINT       => IntegerType
+      case java.sql.Types.TINYINT       => ByteType
```

Comment:
Since `TINYINT` represents an 8-bit unsigned integer value between 0 and 255, wouldn't it be more correct to use `ShortType`?

Reply:
@HyukjinKwon  Thank you for reviewing.

The range of ```TINYINT``` varies by database. It ranges from -128 to 127 for MySQL(https://dev.mysql.com/doc/refman/8.0/en/integer-types.html), H2 (http://www.h2database.com/html/datatypes.html#tinyint_type) and Phoenix (https://phoenix.apache.org/language/datatypes.html#tinyint_type).
It ranges from 0 to 255 for SQL Server (https://docs.microsoft.com/en-us/sql/t-sql/data-types/int-bigint-smallint-and-tinyint-transact-sql?view=sql-server-2017)

I think users can implement ```JdbcDialect``` to override the default JDBC to catalyst type mapping when using the JDBC DataSource. Let me know what you think.

Reply:
Sure that makes sense. However ```JdbcUtils.makeGetter``` does not handle ByteType currently, so if you have a dialect that maps TINYINT to ByteType it will fail with the following exception:
```
java.lang.IllegalArgumentException: Unsupported type tinyint
	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.org$apache$spark$sql$execution$datasources$jdbc$JdbcUtils$$makeGetter(JdbcUtils.scala:502)
	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$$anonfun$org$apache$spark$sql$execution$datasources$jdbc$JdbcUtils$$makeGetters$1.apply(JdbcUtils.scala:379)
	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$$anonfun$org$apache$spark$sql$execution$datasources$jdbc$JdbcUtils$$makeGetters$1.apply(JdbcUtils.scala:379)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.mutable.ArrayOps$ofRef.map(ArrayOps.scala:186)
	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.org$apache$spark$sql$execution$datasources$jdbc$JdbcUtils$$makeGetters(JdbcUtils.scala:379)
	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$$anon$1.<init>(JdbcUtils.scala:340)
```

I will update the PR with with only the change to handle ```ByteType``` in  ```JdbcUtils.makeGetter``` and add a test for this as well.

Reply:
SGTM

Reply:
In that case, I think we don;t need to update migration guide.

Reply:
@twdsilva Could you run [docker-integration-tests](
https://github.com/apache/spark/tree/master/external/docker-integration-tests) manually.

Reply:
@HyukjinKwon  I have updated the PR, please take a look. 
@wangyum Sure, I will run the docker integration tests and post the results.